docs/variables.md

```diff
@@ -25,6 +25,9 @@ For instructions on using variable substitutions see the relevant section of [th
 | `tasks.<taskName>.results.<resultName>[i]` | The ith value of the `Task's` array result. Can alter `Task` execution order within a `Pipeline`.) |
 | `tasks.<taskName>.results['<resultName>'][i]` | (see above)) |
 | `tasks.<taskName>.results["<resultName>"][i]` | (see above)) |
```

Comment:
> Is there another place we mention that [*] only applies to results that have type array as well as can only be used in params of type array
I can add it to the piplines.md's corresponding section.

> Also the fact that it cannot be used in scripts at the moment
It may be missing in current doc, I will check and add it!